src/js/main.js

```diff
@@ -10,17 +10,41 @@
 		obj[key] = src;
 	}
 
-	GameContext = {
+	function displayErrors(elem, errors) {
+		msg = ""
+
+		while (errors && errors.length > 0) {
+			error = errors.pop();
+			msg += (error + "<br>");
+		}
+		
+		elem.innerHTML = msg;
+		elem.display = msg.length > 0 ? elem.display : "none";
+	}
+
+	var errorContent = document.getElementById(errorContentId);
+	
+	var gameContext = {
 		"canvas": null,
 		"engine": null,
 		"gdc": null
 	}
 	
-	var canvas = {canvas: document.getElementById(canvasId)};
-	safeSet(GameContext, canvas, "canvas");
-	TurbulenzEngine = WebGLTurbulenzEngine.create(canvas);
-	safeSet(GameContext, TurbulenzEngine, "engine");
-	var gdc = TurbulenzEngine.createGraphicsDevice({});
-	safeSet(GameContext, gdc, "gdc")
+	try {
+		var canvas = {canvas: document.getElementById(canvasId)};

```

Comment:
Reduce scope of GameContext - it exists purely to provide the low-level engine objects to the classes that will be communicating with them.